fido/fido.py

```diff
@@ -127,11 +135,23 @@ def connectionLost(self, reason):
 
         if (reason.check(_twisted_web_client().ResponseDone) or
                 reason.check(_twisted_web_client().PotentialDataLoss)):
+
+            content_encoding = self.response.headers.getRawHeaders(
+                'Content-Encoding', [])
+            body = self.buffer.getvalue()
+
+            if 'gzip' in content_encoding and self.gzip_enabled:
+                try:
+                    body = zlib.decompress(body, GZIP_WINDOW_SIZE)
+                except zlib.error as e:
+                    self.finished.errback(Failure(GzipDecompressionError(e)))
```

Comment:
Correct me if I'm wrong but I think we should use: 
`self.finished.errback(Failure(exc_type=GzipDecompressionError, exc_value=GzipDecompressionError(e), exc_tb=sys.exc_info()[2]))`

to preserve information while reraising GzipDecompressionError from e

[Twisted docs for Failure](https://twistedmatrix.com/documents/current/api/twisted.python.failure.Failure.html)